File: libs/grpc/clients/trackernet.go
package clients

import (
	"fmt"
	"github.com/satont/tsuwari/libs/grpc/generated/trackernet"
	"log"

	"github.com/satont/tsuwari/libs/grpc/servers"
	"google.golang.org/grpc"
	"google.golang.org/grpc/balancer/roundrobin"
	"google.golang.org/grpc/credentials/insecure"
)

func NewTrackernet(env string) trackernet.TrackernetClient {
	serverAddress := createClientAddr(env, "trackernet", servers.TRACKERNET_SERVER_PORT)

	conn, err := grpc.Dial(
		serverAddress,
		grpc.WithTransportCredentials(insecure.NewCredentials()),
		grpc.WithDefaultServiceConfig(fmt.Sprintf(`{"loadBalancingConfig": [{"%s":{}}]}`, roundrobin.Name)),
	)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	c := trackernet.NewTrackernetClient(conn)

	return c
}

File: libs/grpc/clients/trackernet.ts
import { ChannelCredentials, createChannel, createClient } from 'nice-grpc';

import { TrackernetClient, TrackernetDefinition } from '../generated/trackernet/trackernet.js';
import { PORTS } from '../servers/constants.js';
import { CLIENT_OPTIONS, createClientAddr, waitReady } from './helper.js';

export const createTrackernet = async (env: string): Promise<TrackernetClient> => {
	const channel = createChannel(
		createClientAddr(env, 'trackernet', PORTS.TRACKERNET_SERVER_PORT),
		ChannelCredentials.createInsecure(),
		CLIENT_OPTIONS,
	);

	await waitReady(channel);

	const client = createClient(TrackernetDefinition, channel);

	return client as any;
};
